chore(blog): post with status update for january 2021 (#185)
asyncapi/website: pages/blog/january-2021-at-asyncapi.md

---
title: "January 2021 at AsyncAPI"
date: 2021-02-11T06:00:00+01:00
type: Communication
tags:
  - status
cover: /img/posts/january-2021-at-asyncapi/cover.webp
authors:
  - name: Lukasz Gornicki
    photo: /img/avatars/lpgornicki.webp
    link: https://twitter.com/derberq
    byline: AsyncAPI Maintainer and Dev Comm Keeper
excerpt: 'I always thought January is a month of a slow start of a new year. I could not have been more wrong. No time to rest, no slow down here.'
featured: true
---

> A new year always comes with New Year's resolution, right? Folks from Bump.sh came up with the best idea you could get this year. Don't you dare to fail this time :smiley:

Reviewer: ```suggestion
A new year always comes with New Year's resolution, right? Folks from [Bump.sh](https://www.bump.sh) came up with the best idea you could get this year. Don't you dare to fail this time :smiley:
```

<TwitterTweetEmbed
  tweetId='1346374069877428225'
  options={{
    cards: 'hidden',
    width: 200,

>     maxWidth: 400

Reviewer: This tweet looks super small (narrow). I found out that changing the CSS makes it a bit better. Have a look.

##### Before:
![Captura de pantalla 2021-02-11 a las 14 14 41](https://user-images.githubusercontent.com/242119/107641120-7ed72580-6c73-11eb-880f-624b4981ed6e.png)

##### After:
![Captura de pantalla 2021-02-11 a las 14 16 30](https://user-images.githubusercontent.com/242119/107641304-bfcf3a00-6c73-11eb-85cf-6db2a803bba3.png)

##### Centering:
![Captura de pantalla 2021-02-11 a las 14 16 10](https://user-images.githubusercontent.com/242119/107641263-b219b480-6c73-11eb-8b9a-e0f72780eb89.png)

Author: @fmvilas how about now?

Reviewer: Beautiful!

<TwitterTweetEmbed
  tweetId='1357262834200117250'
  options={{
    cards: 'hidden',
    width: 200,
    maxWidth: 400
  }}
/>

## Community continues to grow

My [last post](/blog/2020-summary) that summarizes the year 2020 was pretty clear that the community's size grew a lot. Well, it is January, and we see that not much has changed. We keep growing:
- we went over 1k users on Slack. [Join now](https://www.asyncapi.com/slack-invite/)
- we went over 1.5k followers on Twitter. [Follow us](https://twitter.com/AsyncAPISpec/) to be up to date with the latest news in the project
- we went over 1.4k stars on GitHub. If you like the project, [express it](https://github.com/asyncapi/asyncapi/stargazers)
- we had several issues solved by the community members from different companies, including few new features. For more details, read [Feature releases and community-driven changes](#feature-releases-and-community-driven-changes) section

Last but not least, we beat the record of people joining our public meetings :rocket:

<TwitterTweetEmbed
  tweetId='1356949862449893376'
  options={{
    width: 200,
    maxWidth: 400
  }}
/>

## RapidAPI Developer Survey

> RapidAPI released the results of their [developer survey](https://rapidapi.com/developer-survey/). Reading it at the beginning of the year is like drinking a strong coffee in the morning - you can a good kick of positive energy for the rest of the year.

Reviewer: ```suggestion
RapidAPI released the results of their [developer survey](https://rapidapi.com/developer-survey/). Reading it at the beginning of the year is like drinking a strong coffee in the morning - you get a good kick of positive energy for the rest of the year.
```

Spoiler alert -> number of developers using AsyncAPI in production tripled in 2020.

## Feature releases and community-driven changes

- [Pavel Bodiachevskii](https://twitter.com/PBodiachevskii) continues his hard work on **asyncapi-java**. Wait, it is finally not **asyncapi-java** anymore. Thanks to a suggestion from [James Higginbotham](https://twitter.com/launchany) it is all about [jasyncapi](https://github.com/asyncapi/jasyncapi/) now. Maven, Gradle and IntelliJ plugins are not published as a preview release under official AsyncAPI accounts. Please give them a try and share your feedback.
- React component:

>   - shows [required flag for required props](https://github.com/asyncapi/asyncapi-react/releases/tag/v0.18.0) thnaks to [c-pius](https://github.com/c-pius) from [SAP](https://twitter.com/SAP)

Reviewer: ```suggestion
  - shows [required flag for required props](https://github.com/asyncapi/asyncapi-react/releases/tag/v0.18.0) thanks to [c-pius](https://github.com/c-pius) from [SAP](https://twitter.com/SAP)
```

## TypeScript NATS template

We have a [new template](https://github.com/asyncapi/ts-nats-template) available. You can use this template to generate [NATS](https://nats.io/) client based on the AsyncAPI document for Node.js. Interesting fact: it is already using the new [React render engine](https://github.com/asyncapi/generator/blob/master/docs/authoring.md#react) from the AsyncAPI Generator.

## Next major feature is data model generation

We again invest big-time in the Generator. This time, it is all about making it super easy to generate a data model for templates, so the template developer doesn't waste much time on templating it and can focus on the template's main logic. In other words, it is all about enabling faster template development. Our [progress](https://shapeup.asyncapi.io/cycles/6301193) looks good, and it seems like at the end of February, we should already have something that you can start using.

Feel free to share your thoughts in [this](https://github.com/asyncapi/shape-up-process/issues/43) issue.

## Refactoring of our CI/CD

> GitHub Actions is what powers our CI/CD. It is a great tool that you configure through a file stored in a repository. Things are just getting more complex when you want to use them in an organization with around 40 repositories (and growing). That is not a post about our internal organizational challenges and GitHub Actions limitations, so I will not bother you with details. The most important is to share that we are managing our GitHub workflows like a pro and if you are interested in more details, contact us.

Reviewer: ```suggestion
GitHub Actions is what powers our CI/CD. It is a great tool that you configure through a file stored in a repository. Things are just getting more complex when you want to use them in an organization with around 40 repositories (and growing). This is not a post about our internal organizational challenges and GitHub Actions limitations, so I will not bother you with details. The most important is to share that we are managing our GitHub workflows like a pro and if you are interested in more details, contact us.
```

So what changed that is meaningful for our community:
- We have two new channels in Slack workspace:
  - #github-releases where you get information about all the releases from all the repositories
  - #github-new-issues-prs where you get information about all new issues and PRs
- Whenever we have a major or a minor release in any repository, our bot automatically tweets about it
- All pull requests are now tested against Linux, MacOS, and Windows. For you, this means that we fixed a lot of bugs in tests and configurations that were blocking Windows users from smooth contributions

> - Not used to Conventional Commits specification? now all pull requests change dedicated check that lints your pull request titles and gives hints what you should fix

Reviewer: ```suggestion
- Not used to Conventional Commits specification? now all pull requests have a dedicated check that lints your pull request titles and gives hints what you should fix
```

## Good read

- [AsyncAPI and OpenAPI: an API Modeling Approach](https://engineering.salesforce.com/asyncapi-and-openapi-an-api-modeling-approach-db9873695910) by [Antonio Garrote](https://twitter.com/antoniogarrote)
- [API adoption is on the rise across all industries](https://www.helpnetsecurity.com/2021/02/05/api-adoption-2021/)
- [How Microcks Can Speed-Up Your AsyncAPI Adoption - Part 2](/blog/microcks-asyncapi-part2) by [Laurent Broudoux](https://twitter.com/lbroudoux)
- [Simplify code generation with React](/blog/react-as-generator-engine) by [Jonas Lagoni](https://www.linkedin.com/in/jonas-terp-lagoni-85b027b9/) and [Maciej Urbańczyk](https://www.linkedin.com/in/maciej-urba%C5%84czyk-909547164/)

> Thank was an exhausting month, but looking at what is happening around the project, you feel it was worth it. Let us see what February brings.

Reviewer: ```suggestion
That was an exhausting month, but looking at what is happening around the project, you feel it was worth it. Let us see what February brings.
```
